File: easy/python/is_palindrome.py
def reverseStr(str):
  counter = 0;
  reversedStr = ''
  
  while(counter < len(str)):
    counter += 1
    reversedStr += str[len(str) - counter]
  return reversedStr
  
def isPalindrome(str):
	reversedStr = reverseStr(str);
	return str.lower() == reversedStr.lower()

print(isPalindrome('Ana'));	# True
print(isPalindrome('cat'));	# False
#	O(n)

File: dart/easy/isPalindrome.dart
class CheckIfPalindrome {
  static String reverseStr(String str) {
    String reversed = '';
    int counter = 0;

    while (counter < str.length) {
      counter++;
      reversed += str[str.length - counter];
    }
    return reversed;
  }

  bool isPalindrome(String str) {
    var reversed = reverseStr(str);
    return str.toLowerCase() == reversed.toLowerCase();
  }
}
